src/Init/Data/Format/Basic.lean

```diff
@@ -17,19 +17,20 @@ inductive Format.FlattenBehavior where
 
 open Format
 
-inductive Format where
-  | nil                 : Format
-  | line                : Format
-  | text                : String → Format
-  | nest (indent : Int) : Format → Format
-  | append              : Format → Format → Format
-  | group               : Format → (behavior : FlattenBehavior := FlattenBehavior.allOrNone) → Format
+inductive Format (τ := Nat) where
+  | nil {}              : Format τ
+  | line {}             : Format τ
```

Comment:
Why the `{}`? It looks like in (almost?) all cases `τ` can be inferred.

Reply:
There are quite a few occurrences where it fails due to instances like `Coe Format MessageData`. Resolution will not assign the MVar to elab `Format.nil` as `coe (Format.nil : Format Nat) : MessageData`.

Reply:
Ah, the `Syntax arity` issue again :)